test/common/constants/pipeline-constants.hpp

```diff
@@ -0,0 +1,22 @@
+#include "src/pipeline/pipeline.hpp"
+
+#define INIT_PIPELINE(I, O, pipeline, stages) Pipeline<I, O> pipeline(stages)
+
+namespace found {
+
+/**
+ * Int->Char Pipeline
+ */
+#define INIT_INT_TO_CHAR_PIPELINE(pipeline, stages) INIT_PIPELINE(int, char, pipeline, stages)
+#define INIT_DOUBLE_TO_STRING_PIPELINE(pipeline, stages) INIT_PIPELINE(double, std::string, pipeline, stages)
+#define INIT_CHAR_TO_DOUBLE_PIPELINE(pipeline, stages) INIT_PIPELINE(char, double, pipeline, stages)
+
```

Comment:
For the below, either:
- convert all these variables into single variables (instead of arrays)
- add more entries, and in your test cases, vary test_set

Also, move these variables to common file, making sure to include the header guard (name it COMMON_H)